src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -328,21 +328,21 @@ private IndexedNode(int bitmap, List<AbstractNode<K, V>> subNodes) {
             int bit = mask;
             int count = 0;
             List<AbstractNode<K, V>> sub = subNodes;
-            List<AbstractNode<K, V>> arr = List.empty();
+            final Object[] arr = new Object[BUCKET_SIZE];
```

Comment:
This line brought me to the question: why we aren't using `T[]` instead of `Object[]`?